packages/flutter/lib/src/widgets/image.dart

```diff
@@ -645,10 +732,18 @@ class _ImageState extends State<Image> {
 
   void _handleImageChanged(ImageInfo imageInfo, bool synchronousCall) {
     setState(() {
+      _chunkEvent = null;
       _imageInfo = imageInfo;
     });
   }
 
+  void _handleImageChunkLoaded(ImageChunkEvent event) {
+    assert(widget.imageLoadingBuilder != null);
+    setState(() {
+      _chunkEvent = event;
```

Comment:
this is REALLY expensive. It forces a rebuild for every frame even if the builder doesn't actually return anything that is affected by this information.

Reply:
But why provide a builder at all if you're not affected by this information?  This information is the essence of why you provided a builder.  And yes, it'll mean that you rebuild every time bytes come across the wire while the image is loading, which is likely every frame.

Reply:
you may wish to show an indeterminate progress indicator while downloading. Or the word "LOADING". Or alternative text in an inline widget.